### ISSUE TITLE
🆕 Software Suggestion | FreeTube

### ISSUE THREAD

**Issue by Kryptor-Software** [privacytools/privacytools.io#2078]
## Basic Information

**Name:** FreeTube
**Category:** Video Platforms - Frontends and Proxies?
**URL:** https://freetubeapp.io/

## Description
From the website:
> FreeTube is a YouTube client for Windows, Mac, and Linux built around using YouTube more privately. You can enjoy your favorite content and creators without your habits being tracked. All of your user data is stored locally and never sent or published to the internet. FreeTube grabs data by scraping the information it needs (with either local methods or by optionally utilizing the Invidious API). With many features similar to YouTube, FreeTube has become one of the best methods to watch YouTube privately on desktop.

## Why I am making the suggestion
Invidious is not a stable alternative to YouTube anymore. Several instances are completely broken/unreliable, some are limited to 720p without an account, and certain instances have got slower to the point that it's not possible to watch a video in HD without buffering (e.g. https://invidious.snopyta.org/ for me). In contrast, the new release of FreeTube (version 0.8.0 Beta) has been significantly more stable than previous versions, and I can play videos in HD without any buffering. It's also feature rich and still in development.

The software is free and open source, there are releases for Windows, macOS, and Linux distros, and the developer seems like a great guy who puts a lot of effort into maintaining the software. My only gripe is the lack of stability in previous versions, but I believe that was partly due to the reliance on Invidious. The stability has definitely improved in the new version, and it will hopefully stay far more operational than Invidious.

## My connection with the software
I am a user of the new FreeTube release (version 0.8.0 Beta).

- [x] I will keep the issue up-to-date if something I have said changes or I remember a connection with the software.

**Comment by LongJohn-Silver** [comment]
Although I really like the idea of FreeTube, saying that Invidious alternatives are unstable in comparison with FT is kind of a joke, the application is a mess, it crashes, sometimes it won't let you watch videos, one time it froze my entire PC, which is odd, because NewPipe works great.

**Comment by Kryptor-Software** [comment]
> Although I really like the idea of FreeTube, saying that Invidious alternatives are unstable in comparison with FT is kind of a joke, the application is a mess, it crashes, sometimes it won't let you watch videos, one time it froze my entire PC, which is odd, because NewPipe works great.

Have you tried the newest version? As I mentioned, previous versions lacked stability, but this new version seems to be a lot better. Version 0.8.0 Beta is a complete rewrite, so I suggest you give the software another chance. Furthermore, there's now another person working on development alongside Preston.

I think it's important to realise that Invidious could completely break one day and nobody will fix it because it's written in Crystal, which is not a popular language. At least FreeTube is being maintained, which means even if there are bugs, they'll eventually get looked into.

**Comment by LongJohn-Silver** [comment]
Yeah, sorry, I thought it was the same version I was running. I've been trying it today and yesterday, and it has some major improvements, it is still a bit buggy, but nothing compared to what it was. Seeing it now, I think it totally deserves to be listed.

**Comment by lynn-stephenson** [comment]
This is can be included in the worth mentioning section considering it still in development, and a bit buggy.

**Comment by samuel-lucas6** [comment]
The main bug I've experienced is with the search, but that issue seems to be with another library he's using - [Search isn't working for either API](https://github.com/FreeTubeApp/FreeTube/issues/662). I've also experienced an issue where loading comments temporarily freezes the window, and the video position for certain videos isn't remembered. Some of these things are due to be fixed.

It's still a lot better than Invidious where the search is completely broken (e.g. the snopyta instance), and most instances don't have 1080p. I agree that it should go in the worth mentioning section.

**Comment by LongJohn-Silver** [comment]
I've encountered a bug where the previous videos I was watching started to sound if I reproduce the current video using my space bar, but not if I click on the play button, and also subtitles wreckage. Still it works far better than Invidious where you have to change of instance every 15 minutes to watch anything.

**Comment by wylel** [comment]
> 
> 
> This is can be included in the worth mentioning section considering it still in development, and a bit buggy.

I dont agree, and think it should be placed next to Invidious and heres why.

1. Invidious is also in beta, even though it doesnt have a beta tag (no 1.0 release).
2. It can be argued that Invidious has slowed in development in recent news.
3. Invidious is not bullet proof and has many bugs, and most instances (especially since the main instance isnt up anymore to say "this is the good one") are an absolute mess, if not completely broken.

However, just like all software and site requests, they are normally looked at for their longevity and ability to perform. I have been using FreeTube since around the announcement the Invidious main instance was going to shut down and, although its had its issues, a lot of those were due to the Invidious API. A local API was written to curtail this and has the option to choose it (with a fallback to the Invidious API if it failes) or choose Invidious still, and has worked pretty well since that was implemented.

I highly recommend it up there with Invidious, if not even before it (based on my previous points).

Also, I do not contribute nor am I affiliated with FreeTube, just a happy user.

**Comment by GilgusMaximus** [comment]
> 
> 
> I've encountered a bug where the previous videos I was watching started to sound if I reproduce the current video using my space bar, but not if I click on the play button, and also subtitles wreckage. Still it works far better than Invidious where you have to change of instance every 15 minutes to watch anything.

It would always help us to if all bugs that are not already reported, could be reported via an issue on our repository.


>
>
> The main bug I've experienced is with the search, but that issue seems to be with another library he's using - Search isn't working for either API. I've also experienced an issue where loading comments temporarily freezes the window, and the video position for certain videos isn't remembered. Some of these things are due to be fixed.

This is just the nature of all privacy friendly YouTube clients. If YouTube changes something small about their naming conventions, one new HTML element in their webpage, it can break a range of modules used by these projects. That's why it is important to get reports of major problems quickly at the respective repositories, so that the maintainers or contributors can fix it.

**Comment by LongJohn-Silver** [comment]
I've also found this alternative to Invidious, which I haven't tried thoroughly, but it seems to work quite better than its sibling. I know this probably should go into another issue, but since the discussion seems to be more in the aspect of what should be considered okay in order to be listed, I figured I'll comment about it.
https://sr.ht/~cadence/tube/
https://tube.cadence.moe/
----
> It would always help us to if all bugs that are not already reported, could be reported via an issue on our repository.
Yes, I know, I've been postponing this, excuse me. I'll try to do it today

**Comment by GintokiHub** [comment]
It's also using the invidio.us api. 
SEE: https://safe.succmy.wang/j56ohfi4.png

**Comment by samuel-lucas6** [comment]
> > I've encountered a bug where the previous videos I was watching started to sound if I reproduce the current video using my space bar, but not if I click on the play button, and also subtitles wreckage. Still it works far better than Invidious where you have to change of instance every 15 minutes to watch anything.
> 
> It would always help us to if all bugs that are not already reported, could be reported via an issue on our repository.
> 
> > The main bug I've experienced is with the search, but that issue seems to be with another library he's using - Search isn't working for either API. I've also experienced an issue where loading comments temporarily freezes the window, and the video position for certain videos isn't remembered. Some of these things are due to be fixed.
> 
> This is just the nature of all privacy friendly YouTube clients. If YouTube changes something small about their naming conventions, one new HTML element in their webpage, it can break a range of modules used by these projects. That's why it is important to get reports of major problems quickly at the respective repositories, so that the maintainers or contributors can fix it.

@GilgusMaximus The reason I didn't report these issues was because there were already reports open, but I understand that these types of issues will always occur from time to time due to YouTube changes.

I think @wylel makes a strong case for FreeTube to be listed next to Invidious. We probably need to start opening separate issues for non-FreeTube discussion, but Yotter is looking like it might be a decent Invidious replacement as well.

**Comment by GilgusMaximus** [comment]
Ah okay, I didn't remember this issue to be open. But could be that I missed it in the mass of all the issue coming in daily. Thanks anyways :)

**Comment by ZeikoFr** [comment]
Hi, just posting a support comment for the addition of Freetube on the website, maybe while specifying that the software is in beta but it could allow a flow a new user and more reporting of issues that could lead to enhancement of the software.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
+1 for adding Freetube. Its really improved over the past months. Way more stable, pleasent viewing experience and a ton of great features wouldn't even call it a beta anymore.

**Comment by samuel-lucas6** [comment]
> 
> 
> +1 for adding Freetube. Its really improved over the past months. Way more stable, pleasent viewing experience and a ton of great features wouldn't even call it a beta anymore.

I agree. I haven't really experienced any problems in the last few months. It's also being maintained very well, with consistent updates. It offers a much better experience than any of the Invidious instances.
